inference-engine/src/readers/ir_reader/ie_ir_parser.hpp

```diff
@@ -183,10 +184,20 @@ class V10Parser : public IParser {
 
     class XmlDeserializer : public ngraph::AttributeVisitor {
     public:
-        explicit XmlDeserializer(const pugi::xml_node& node, const Blob::CPtr& weights,
-        const std::unordered_map<std::string, ngraph::OpSet>& opsets,
-        std::unordered_map<std::string, std::shared_ptr<ngraph::Variable>>& variables)
-        : node(node), weights(weights), opsets(opsets), variables(variables) {}
+        /// FIX: why whole class is define here not in src file?
+        enum class io { INPUTS, OUTPUTS };
+
+        using NodeIdToIoIndex = std::map<size_t /*xml node id*/, uint64_t /*body io index*/>;
+        using IoMap = std::map<io, NodeIdToIoIndex>;
+        using NodeIdToIoIndexStack = std::stack<IoMap>;
```

Comment:
Why do we need to have an additional stack for Input/Outputs?
Do we use this stack only for TI/Loop?

Reply:
We need this `static` stack to pass from data from one `XmlDeserializer` instanton to another. 
We have such scenario:
1) start deserialization - create `XmlDeserializer` and pass function to it
2) visit `SubGraphOp` and it attributes
 - visit `body`, which require yet another instance of `XmlDeserializer` (this serialization have to return parameter/result order)
 - visit Input/Output Description (use info came from `body` deserialization about input/outputs (parameter/result) order)
 
 For now I can't find better way to pass by VisitorAPI call stack such information.

Reply:
Hmmm, can we add parent `XMLDeserializer` as an argument when we create `XMLDeserializer` for nodes or maybe share some property?

Reply:
It looks `XmlDeserializer::createNode` create yet another instance of `XmlDeserializer` and from lever of `TI` processing there is the same instance of `XmlDeserializer`. Conclusion: no stack required just single member.